src/clarity.rs

```diff
@@ -137,18 +114,23 @@ pub fn invoke_command(invoked_by: &str, args: &[String]) {
             
             let content: String = friendly_expect(fs::read_to_string(&args[1]),
                                                   &format!("Error reading file: {}", args[1]));
-            
-            let mut db_conn = {
+
+            let mut ast = friendly_expect(parse(&content), "Failed to parse program");
+
+            let contract_analysis = {
                 if args.len() >= 3 {
-                    AnalysisDatabaseConnection::open(&args[2])
+                    // use a persisted marf
+                    let mut marf = friendly_expect(sqlite_marf(&args[2], None), "Failed to open VM database.");
+                    let result = { let mut db = AnalysisDatabase::new(Box::new(&mut marf));
+                                   type_check(&":transient:", &mut ast, &mut db, false) };
+                    marf.commit();
+                    result
                 } else {
-                    AnalysisDatabaseConnection::memory()
+                    let mut memory = friendly_expect(SqliteConnection::memory(), "Could not open in-memory analysis DB");
```

Comment:
Should there be a RAM-only MARF?  Or, a way to instantiate a RAM-only MARF?

Reply:
In the context of analysis, can't we drop the MARF and use a regular HashMap?

Reply:
Oh nvm, we're already doing that 🚀

Reply:
I'm not sure -- for transient execution contexts like this, using a MARF wouldn't change much, and so I'm hesitant to say we should implement a RAM-only MARF. Something like a RAM-only MARF might be useful for an API, but I suspect that the exact requirements of that wouldn't be obvious until the API is implemented, so I think we should hold off.

Reply:
My main concern is code complexity.  If we find ourselves having to maintain two different contract analysis DB implementations -- one with a MARF and one without -- then it would be relatively straightforward to build a RAM-only MARF.  But if you don't think this is a problem for the foreseeable future then I'm fine to leave this as-is :+1:

Reply:
Yeah -- I think there's relatively little added complexity here, since the DB's are just expecting that KeyValueStore trait (and the SQLite implementation of that trait is necessary anyways for the side storage). If we didn't use the generics, we could get rid of the necessity of `Box`ing the KeyValueStore (it could _probably_ be eliminated with an associated type, but my brain-compiler tends to get more bogged down reasoning about associated types than it does with just `Box`ing a trait.